README.md

```diff
@@ -115,13 +115,63 @@ Note that `called` can be used as a chainable method.
 
 #### .with
 
-Assert that a spy has been called with a given argument.
+Assert that a spy has been called with a given argument at least once,
+even if more arguments were provided.
 
 ```js
 expect(spy).to.have.been.called.with('foo');
 spy.should.have.been.called.with('foo');
 ```
 
+Will pass for `spy('foo')`, `spy('foo', 'bar')`, `spy(); spy('foo')`,
+etc.
+
+Can be used for multiple arguments:
+
+```js
+spy('foo');
+spy('bar');
+expect(spy).to.have.been.called.with('bar', 'foo');
+spy.should.have.been.called.with('bar', 'foo');
+```
+
+#### .with.exactly
+
+Similar to .with, but will pass only if the list of arguments is
+exactly the same as the one provided.
+
+```js
+expect(spy).to.have.been.called.with.exactly('foo', 'bar');
+spy.should.have.been.called.with.exactly('foo', 'bar');
+```
+
+Will pass only if `spy('foo', 'bar')` happened, at least once.
+
+Will not pass for `spy('foo')`, `spy('bar')`, `spy('bar');
+spy('foo')`, `spy('foo'); spy('bar')` or `spy('bar', 'foo')`.
+
+Can be used for calls with a single argument too.
+
+#### .always.with
+
+Assert that every time the spy has been called the argument list
+contained the given arguments.
+
+```js
+expect(spy).to.have.been.called.always.with('foo');
+spy.should.have.been.called.always.with('foo');
+```
```

Comment:
Same again here; adding the spy calls to illustrate how these assertions would pass would be great.